[isaactodo] iP

isaactodo opened the pull request:
# 🤖 Sentinel

## 🤔 Why should you use Sentinel?
- It was made by the **best coder** in the world, me
- Has _pretty_ UI :heart_eyes:
- Unofficially endorsed by **SEN Tenz**

> Sentinel software is ~~really good~~ **senceptional**! - SEN Tenzania

## 📖 How to use Sentinel?
1. Download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. Input your tasks.
3. Let Sentinel manage your tasks.
4. Profit. 🤑 

## 💯 Features
- [x] Manages your tasks
- [x] Saves your tasks
- [ ] What else could you ask for?

### Here's a snippet of its amazing `main` method
```java
public static void main(String[] args) {
    Sentinel mySentinel = new Sentinel();
    mySentinel.greet();
    mySentinel.listen();
    mySentinel.goodbye();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

isaactodo committed:
feat: Rename chatbot, greet and exit

isaactodo committed:
feat: Add echo feature to chatbot

isaactodo committed:
feat: Add ability to store and output lists

isaactodo committed:
Add javadocs to public methods

isaactodo committed:
feat: add command parsing and marking system

isaactodo committed:
feat: add todos, events and deadlines

isaactodo committed:
Removed scanner import

isaactodo committed:
fix: removed extra println statement

isaactodo committed:
feat: Add automated text UI testing

isaactodo committed:
fix: Fix automatic testing

isaactodo committed:
Add SentinelException class

isaactodo committed:
TaskList.java: Change to use ArrayList instead of arrays

isaactodo committed:
Add delete command to Sentinel

isaactodo committed:
Fix error when trying to index a task that does not exist

isaactodo committed:
Add javadocs to most public methods

isaactodo committed:
Add test cases for deletion and marking tasks

isaactodo committed:
Add task saving feature

isaactodo committed:
Update .gitignore to ignore data files

isaactodo committed:
Merge branch 'branch-Level-7'

isaactodo committed:
Add date and time support

isaactodo committed:
Merge branch 'branch-Level-8'

isaactodo committed:
Rename FileManager.java to Storage.java

isaactodo committed:
Move most parsing functions to Parser.java

isaactodo committed:
Add Ui.java file

isaactodo committed:
Merge branch 'A-MoreOOP'

isaactodo committed:
Refactor code into packages and add enums to Commands.java

isaactodo committed:
Merge branch 'add-gradle-support'

isaactodo committed:
Add Gradle automation

isaactodo committed:
Change Ui error message to not trigger Gradle error

isaactodo committed:
Add unit tests to Commands, Parser and TaskList

isaactodo committed:
Change gradle main class

isaactodo committed:
Add javadocs to most public methods

isaactodo committed:
Refactor code to comply with Java coding standard

isaactodo committed:
Add command to find tasks

isaactodo committed:
Merge branch 'branch-A-JavaDoc'

isaactodo committed:
Merge branch 'branch-A-CodingStandard'

isaactodo committed:
Merge branch 'branch-Level-9'

isaactodo committed:
Add checkstyle functionality

isaactodo committed:
Add JavaFX to Sentinel

isaactodo committed:
Change implementation for goodbye message

isaactodo committed:
Change code to comply with coding style

isaactodo committed:
Change test code to comply with coding style

isaactodo committed:
Add better UI design to chat bot

isaactodo committed:
Merge branch 'branch-A-Checkstyle'

isaactodo committed:
Fix merge conflict

isaactodo committed:
Refactor merged code to comply with coding standard

JumpyJay commented on src/main/java/sentinel/Sentinel.java:
```diff
@@ -0,0 +1,247 @@
+
+    /**
+     * Makes Sentinel mark the given task number as done.
+     *
```
could perhaps add @throws and elaborate on when could it be thrown

isaactodo replied:
Oops, forgot to add @throws

JumpyJay commented on src/main/java/sentinel/Sentinel.java:
```diff
@@ -0,0 +1,247 @@
+
+    /**
+     * Makes Sentinel mark the given task number as undone.
+     *
```
Similar to line67, could perhaps add @throws and elaborate on when could it be thrown

JumpyJay commented on src/main/java/sentinel/Sentinel.java:
```diff
@@ -0,0 +1,247 @@
+    /**
+     * Makes Sentinel add the given todo to the task list.
+     *
+     * @param description Description of the todo to be added.
```
Might be some spelling error on "description Description" perchance, same applies to line 107 & 116

JumpyJay commented on src/main/java/sentinel/Sentinel.java:
```diff
@@ -0,0 +1,247 @@
+    /**
+     * Makes Sentinel add the given event to the task list.
+     *
+     * @param description Description of the event to be added.
```
Would be helpful if '@param' was made clearer, the same @param for addEvent and addDeadline (on line 116) makes it seem like these two task have the same params

isaactodo replied:
Noted!

JumpyJay reviewed (commented):
LGTM, just some nits to fix :P

isaactodo reviewed (commented):
Replying to comments

isaactodo committed:
Add assertions to various points in the code

There are no assertions in the code.

Assertions can be used to define assumptions about the program state
so that the runtime can verify them.

Let's
* add assertions to Sentinel methods to check if task list exists
* add assertions to Storage file load() method to check if the file
was created correctly
* add assertions to TaskList methods to check if the task to modify
exists

Assertions are used this way to verify that the assumptions made
about the program are correct and to catch potential bugs.

See
https://nus-cs2103-ay2425s1.github.io/website/schedule/
week5/topics.html#w5-6-assertions

isaactodo committed:
Refactor code to improve code quality

Some methods are longer than 30 LoC, minor presence of arrowhead
style code and weak SLAP.

These factors decrease the readability of the code, in turn
reducing the code quality.

Let's
* use Extract Method to reduce the LoC of some methods in Sentinel
class and to increase SLAP.
* split parseStringToTask in Parser class into several methods to
reduce the length of the method and to increase SLAP.
* extract out a loadFile method from Storage.load() to reduce the
overall length of the method.

isaactodo committed:
Merge pull request #1 from isaactodo/branch-A-Assertions

Add assertions to various points in the code

isaactodo committed:
Resolve merge conflict

isaactodo committed:
Merge pull request #2 from isaactodo/branch-A-CodeQuality

Refactor code to improve code quality

isaactodo committed:
Setup Continuous Integration

isaactodo committed:
Add client management support

isaactodo committed:
Merge branch 'branch-D-Clients'

isaactodo committed:
Fix method header comments to comply with coding standard

isaactodo committed:
Fix automated text ui testing not passing workflow

isaactodo committed:
Remove references to duke in JavaFX files

isaactodo committed:
Add screenshot and user guide

isaactodo committed:
Fix user guide
